aws_lambda_builders/workflows/__init__.py

```diff
@@ -11,3 +11,4 @@
 import aws_lambda_builders.workflows.java_maven
 import aws_lambda_builders.workflows.dotnet_clipackage
 import aws_lambda_builders.workflows.custom_make
+import aws_lambda_builders.workflows.rust_cargo
```

Comment:
Just to call it our for those less familiar. This _is_ the architecture hook that I understood existed for the purposes of updating the workflow registry

The feedback given on twitter may have indicated that this is not the way this repo is supposed to be extended

https://twitter.com/alexwwood/status/1281413395585961984?s=19

Please advise

Reply:
@softprops I will bring this up with the team today and chat with Alex. I think there may be some kind of miss understanding going on here.

My view:
We should be enabling things like Rust through cargo, just because it is a 'provided' runtime doesn't mean we can't add a builder for it. Once we add a builder here, SAM CLI will need to be updated to 'discover' the `cargo.toml` file and run this builder. 

What Alex might be referring to here is that this will work for `sam build` but not for `sam build --use-container`, since the build container for provided may not have what it needs to build (cargo/rust). But since rust is compiled to a binary (and assume rust can cross compile), this shouldn't block customers from building. We will probably want to add a further check on the `sam build --use-container` to prompt that rust/cargo cannot be built with a container.

I am going to start reviewing this and providing feedback now. I will also sync up with the team to make sure we are all on the same page going forward.

Reply:
Thanks.

> Once we add a builder here, SAM CLI will need to be updated to 'discover' the cargo.toml file and run this builder.

I'm aware and planned to do that work. This seemed like the the part that needed to come first. I already familiarized myself with how the two projects work despite python not being my my strong suit :) as mentioned I had plotted out the time it would take to do both but had not factored in the review feedback latency. I didn't have a baseline to compare it too up front.

> but not for sam build --use-container, since the build container for provided may not have what it needs to build (cargo/rust)

Understand able the same is true for go I think there's a built-in hook I saw for for communicating that I could leverage. That said I should note I've been maintaining the serverless plugin for rust for years and am intimately familiar with the docker option. We use lambdaci projects provided builder with rustup preinstalled. I'm really starting to favor local builds after. That experience but if you need it, I know how to make the docker option work